modules/insticatorBidAdapter.js

```diff
@@ -12,35 +12,53 @@ const USER_ID_COOKIE_EXP = 2592000000; // 30 days
 const BID_TTL = 300; // 5 minutes
 const GVLID = 910;
 
-const isSubarray = (arr, target) => {
+const filterArray = (arr, target) => {
   if (!isArrayOfNums(arr) || arr.length === 0) {
-    return false;
+    return [];
   }
   const targetSet = new Set(target);
-  return arr.every(el => targetSet.has(el));
+  return arr.filter(el => targetSet.has(el));
 };
 
+const VIDEO_NUM_ARRAY_PARAMS = {
+  'protocols': [2, 3, 5, 6, 7, 8],
+  'api': [1, 2, 3, 4, 5, 6, 7],
+  'playbackmethod': [1, 2, 3, 4, 5, 6, 7],
+  'delivery': [1, 2, 3],
+  'battr': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17],
+}
+
 export const OPTIONAL_VIDEO_PARAMS = {
   'minduration': (value) => isInteger(value),
   'maxduration': (value) => isInteger(value),
-  'protocols': (value) => isSubarray(value, [2, 3, 5, 6, 7, 8]), // protocols values supported by Inticator, according to the OpenRTB spec
+  'protocols': (value) => isArrayOfNums(value), // protocols values supported by Inticator, according to the OpenRTB spec
   'startdelay': (value) => isInteger(value),
   'linearity': (value) => isInteger(value) && [1].includes(value),
   'skip': (value) => isInteger(value) && [1, 0].includes(value),
   'skipmin': (value) => isInteger(value),
   'skipafter': (value) => isInteger(value),
   'sequence': (value) => isInteger(value),
-  'battr': (value) => isSubarray(value, [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17]),
+  'battr': (value) => isArrayOfNums(value),
   'maxextended': (value) => isInteger(value),
   'minbitrate': (value) => isInteger(value),
   'maxbitrate': (value) => isInteger(value),
-  'playbackmethod': (value) => isSubarray(value, [1, 2, 3, 4]),
+  'playbackmethod': (value) => isArrayOfNums(value),
   'playbackend': (value) => isInteger(value) && [1, 2, 3].includes(value),
-  'delivery': (value) => isSubarray(value, [1, 2, 3]),
+  'delivery': (value) => isArrayOfNums(value),
   'pos': (value) => isInteger(value) && [0, 1, 2, 3, 4, 5, 6, 7].includes(value),
-  'api': (value) => isSubarray(value, [1, 2, 3, 4, 5, 6, 7]),
+  'api': (value) => isArrayOfNums(value),
 };
 
+const ORTB_SITE_FIRST_PARTY_DATA = {
+  'cat': v => Array.isArray(v) && v.every(c => typeof c === 'string'),
+  'sectioncat': v => Array.isArray(v) && v.every(c => typeof c === 'string'),
+  'pagecat': v => Array.isArray(v) && v.every(c => typeof c === 'string'),
+  'search': v => typeof v === 'string',
+  'mobile': v => isInteger(),
+  'content': v => typeof v === 'object',
+  'keywords': v => typeof v === 'string',
```

Comment:
even though this is under site in prebid js, we can use this for both site & content in bid requests on the exchange right?

Reply:
Also we are doing any validation on `content` object from here. We can do that in exchange. @rohanInsticator